src/helpers/search-data.ts

```diff
@@ -88,7 +94,74 @@ export const indexAndSearch = async (
   // Index the data
   const index = indexData(data);
 
+  let existingData;
+
+  if (
+    index &&
+    index !== null &&
+    typeof index !== undefined &&
+    index.length > 0
+  ) {
+    if (!fs.existsSync(pathToSearchIndex)) {
+      console.log("File does not exist");
+      await saveJson(index);
+    } else {
+      console.log("File exists");
+
+      try {
+        const data = fs.readFileSync(pathToSearchIndex, "utf8");
+        existingData = JSON.parse(data);
+        const mergedData = mergeData(existingData.entries, index);
+        await saveJson(mergedData);
+      } catch (error) {
+        console.log("Failed to parse JSON", error);
+      }
+    }
+  } else {
+    console.log("No data to index");
+  }
+
   // Search the data
-  const searchResults = searchData(index, query);
+  const searchResults = searchIndexForData(index, query);
+
   return { searchResults, totalSearchResults: searchResults.length };
 };
+
+const saveJson = async (data: any) => {
+  if (!data) return console.log("No data to save");
+  const dataToSave = { entries: data };
+  const json = JSON.stringify(dataToSave, null, 2);
+  await writeFile(pathToSearchIndex, json, "utf8")
+    .then(() => {
+      console.log("JSON file has been saved.");
+    })
+    .catch((err) => {
+      console.log("An error occurred while writing JSON Object to File.");
+      console.log(err);
+    });
+};
+
+const mergeData = (existingData: any, newData: any) => {
+  const idSet = new Set();
+  existingData.forEach((item: any) =>
+    idSet.add(item.title + item.authors + item.summary + item.link + item.path)
+  );
+
+  // Use a set to avoid duplicates
+  const mergedData = [...existingData];
+
+  newData.forEach((newItem: any) => {
+    const id =
+      newItem.title +
```

Comment:
We want the date in here too I think

Reply:
This is isn't used in the search data. This is just to check for uniqueness of data when we update the JSON. So we don't need it here but here -> https://github.com/bitcoinsearch/tldr/pull/6/files#diff-437bd2d20487050ec598481bff02c73c03647954af894e2e799fd29fea42da71R44
I'll add that